libs/message-coder/src/byte_array_with_length_prefix.ts

```diff
@@ -80,12 +80,14 @@ class FixedLengthByteArrayCoder extends BaseCoder<Uint8Array> {
   constructor(private readonly length: number) {
     super();
 
-    const bytesRequired = Math.ceil(Math.log2(length + 1) / BITS_PER_BYTE);
+    const bytesRequired = Math.max(
+      Math.ceil(Math.log2(length + 1) / BITS_PER_BYTE),
+      1
```

Comment:
Rather than clapping to 1, just assert that `length > 0`?

Reply:
When `length = 1`, `bytesRequired = 0` based on the math, so we still need to do something here.

We could further make the assertion that `length !== 0`, because why would you be encoding a fixed length array of length 0, but that's not addressing any flakiness.

Reply:
Doesn't `length=1` result in `bytesRequired=1` even without `Math.max`? Evaluating in browser console:

```
> const length = 1;
< undefined
> const BITS_PER_BYTE = 8;
< undefined
> Math.ceil(Math.log2(length + 1) / BITS_PER_BYTE)
< 1
```

Separately, I also agree we can assert `length !== 0` since encoding fixed length arrays of length 0 has ~undefined behavior.

Reply:
Ah, you're right, I was mistaken. I'll switch to making that assertion.

Reply:
Fixed it. I don't think the `assert` is even necessary as it is captured by the `default` case of the `switch` block, with accurate logging.

Reply:
Ah true, without `Math.max`, `bytesRequired = 0` when `length = 0`